WooCommerce/Classes/ViewRelated/Blaze/AdDestination/BlazeAdDestinationSettingViewModel.swift

```diff
@@ -0,0 +1,113 @@
+import Foundation
+
+final class BlazeAdDestinationSettingViewModel: ObservableObject {
+    enum DestinationURLType {
+        case product
+        case home
+    }
+
+    var productURL: String
+    var homeURL: String
+
+    @Published var selectedDestinationType: DestinationURLType
+
+    @Published var parameters: [BlazeAdURLParameter]
+
+    // Text to be shown on the view for remaining available characters for custom added parameters.
+    var remainingCharactersLabel: String {
+        let remainingCharacters = calculateRemainingCharacters()
+        let lengthText = String.pluralize(remainingCharacters,
+                                             singular: Localization.parameterCharactersLimit.singular,
+                                             plural: Localization.parameterCharactersLimit.plural)
+        return String(format: lengthText, remainingCharacters)
+    }
+
+    // Text to be shown on the view for the final ad campaign URL including parameters, if any.
+    var finalDestinationLabel: String {
+        return String(format: Localization.finalDestination, buildFinalDestinationURL())
+    }
+
+    init (productURL: String,
+          homeURL: String,
+          selectedDestinationType: DestinationURLType = .product,
+          parameters: [BlazeAdURLParameter] = [BlazeAdURLParameter(key: "key1", value: "value1")]) {
+        self.productURL = productURL
+        self.homeURL = homeURL
+        self.selectedDestinationType = selectedDestinationType
+        self.parameters = parameters
+    }
+
+    func setDestinationType(type: DestinationURLType) {
+        selectedDestinationType = type
+    }
+
+    private func buildParameterString() -> String {
+        var parameterString = ""
+        for parameter in parameters {
+            // In URL format, the parameter is written such as "key=value".
+            parameterString += parameter.key + "=" + parameter.value
+
+            // If it's not the last parameter, add an ampersand.
+            if parameter != parameters.last {
+                parameterString += "&"
+            }
+        }
+
+        return parameterString
+    }
```

Comment:
Suggestion: this can be simplified further

```suggestion
    private var parameterString: String {
        parameters.map { $0.key + "=" + $0.value }.joined(separator: "&")
    }
```

Reply:
Very nice! Love the use of joined there.